beacon-chain/rpc/beacon/attestations.go

```diff
@@ -134,44 +146,40 @@ func (bs *Server) ListIndexedAttestations(
 			NextPageToken:       strconv.Itoa(0),
 		}, nil
 	}
-
-	// We use the retrieved committees for the epoch to convert all attestations
+	// We use the retrieved committees for the block root to convert all attestations
 	// into indexed form effectively.
+	mappedAttestations := mapAttestationsByBlockRoot(ctx, attsArray)
 	indexedAtts := make([]*ethpb.IndexedAttestation, numAttestations, numAttestations)
-	for i := 0; i < len(atts); i++ {
-		att := atts[i]
-		epoch := helpers.SlotToEpoch(att.Data.Slot)
-		attState, err := bs.StateGen.StateByRoot(ctx, bytesutil.ToBytes32(att.Data.BeaconBlockRoot))
-		if err != nil {
-			return nil, status.Errorf(
-				codes.Internal,
-				"Could not retrieve state for attestation data block root %v: %v",
-				att.Data.BeaconBlockRoot,
-				err,
-			)
+	attIndex := 0
+	for blockRoot, atts := range mappedAttestations {
+		var attState *stateTrie.BeaconState
+		if !featureconfig.Get().DisableNewStateMgmt {
+			attState, err = bs.StateGen.StateByRoot(ctx, blockRoot)
+		} else {
+			attState, err = bs.BeaconDB.State(ctx, blockRoot)
```

Comment:
I have a concern about this.  Without new state mgmt, every state between genesis to last finalized epoch gets pruned. What happens if `attState` is nil.

Reply:
running it in runtime to test

Reply:
it returns an ugly error . but doesnt panic. will return error if stategen disabled